### PR TITLE
sl-243: Steps display of completed information

### PR DESCRIPTION
**Pull request by Alexandr-mas**
https://user-images.githubusercontent.com/84984785/230967715-4463a0ac-7b82-4d01-a14d-0f0245fec866.mp4

It is made that when entering information in the form and moving between steps, the information is saved if the user wants to return to the passed steps.

### CODE REVIEW

File: frontend/src/pages/start/libs/components/industry-form/industry-form.tsx [hunk 1]
```diff
@@ -1,29 +1,35 @@
...
 
-const IndustryForm: React.FC<Properties> = ({ onSubmit }: Properties) => {
+const IndustryForm: React.FC<Properties> = ({
+  onSubmit,
+  siteInfo,
+}: Properties) => {
+  const [industry] = useState(siteInfo.industry || '');
   const { control, errors, handleSubmit } = useAppForm<SiteCreateStepIndustry>({
-    defaultValues: DEFAULT_STEP_PAYLOAD,
+    defaultValues: { industry: siteInfo.industry || '' },
```
Comment (andanf-e) [inline frontend/src/pages/start/libs/components/industry-form/industry-form.tsx]:
```suggestion
    defaultValues: { ...DEFAULT_STEP_PAYLOAD, industry: stepInfo.industry },
```

File: frontend/src/pages/start/libs/components/industry-form/industry-form.tsx [hunk 2]
```diff
@@ -1,29 +1,35 @@
...
   onSubmit: (sitePayload: Partial<SiteCreateRequestDto>) => void;
+  siteInfo: SiteCreateRequestDto;
 };
 
-const IndustryForm: React.FC<Properties> = ({ onSubmit }: Properties) => {
+const IndustryForm: React.FC<Properties> = ({
+  onSubmit,
+  siteInfo,
+}: Properties) => {
+  const [industry] = useState(siteInfo.industry || '');
```
Comment (andanf-e) [inline frontend/src/pages/start/libs/components/industry-form/industry-form.tsx]:
```suggestion

```

File: frontend/src/pages/start/libs/components/industry-form/industry-form.tsx [hunk 3]
```diff
@@ -34,7 +40,7 @@ const IndustryForm: React.FC<Properties> = ({ onSubmit }: Properties) => {
         <Input
           type="text"
           label="Project industry"
-          placeholder="Industry"
+          placeholder={industry || 'Industry'}
```
Comment (andanf-e) [inline frontend/src/pages/start/libs/components/industry-form/industry-form.tsx]:
```suggestion
          placeholder="Industry"
```

File: frontend/src/pages/start/libs/components/industry-form/industry-form.tsx [hunk 4]
```diff
@@ -1,29 +1,35 @@
...
 
   const handleFormSubmit = useCallback(
     (event_: React.BaseSyntheticEvent): void => {
-      void handleSubmit(onSubmit)(event_);
+      void handleSubmit((data) => {
+        onSubmit({ ...siteInfo, ...data });
+      })(event_);
     },
-    [handleSubmit, onSubmit],
+    [handleSubmit, onSubmit, siteInfo],
```
Comment (andanf-e) [inline frontend/src/pages/start/libs/components/industry-form/industry-form.tsx]: Turn it back

**Review by andanf-e — COMMENTED** [review]
And this is for all forms - all that you need is just take Site Creation state and pass the current form default value part from it

File: frontend/src/pages/start/libs/components/voice-tone-form/voice-tone-form.tsx [hunk 5]
```diff
@@ -10,11 +11,15 @@ import styles from './styles.module.scss';
...
   onSubmit: (sitePayload: Partial<SiteCreateStepTone>) => void;
+  siteInfo: SiteCreateRequestDto;
 };
 
-const VoiceToneForm: React.FC<Properties> = ({ onSubmit }: Properties) => {
+const VoiceToneForm: React.FC<Properties> = ({
+  onSubmit,
+  siteInfo,
+}: Properties) => {
   const { control, errors, handleSubmit } = useAppForm({
```
Comment (what1s1ove) [inline frontend/src/pages/start/libs/components/voice-tone-form/voice-tone-form.tsx]:
```suggestion
  const { control, errors, handleSubmit } = useAppForm<SiteCreateStepTone>({
```

**Review by what1s1ove — APPROVED** [review]

**Pull request merged**